extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -67,6 +68,43 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(StateProperty.PAGE_SIZE, () => {
+      this.onPageSizeStateUpdate_();
+    });
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @private
+   */
+  shoudTagFlip_() {
+    const {offsetLeft, offsetWidth} = this.element;
```

Comment:
For AMP components (`amp-story`, `amp-story-shopping-tag`, etc), please get the cached measurements using `el.getLayoutBox()` instead of re-measuring

Reply:
I am getting this when I try to use `. getLayoutBox()` , are you sure you want me to use it?

Forbidden: ".getLayoutBox". getLayoutSize/Box APIs are being deprecated. Please contact the @ampproject/wg-performance for questions.eslint(local/no-forbidden-terms)

Reply:
If ok to make async, you can make the measure nonblocking via using either InOb or ResOb:

https://github.com/ampproject/amphtml/blob/a993fd73d200a544029296d9e40e66d33c6b5718/src/core/dom/layout/intersection-no-root.js#L69-L81

Reply:
We are using offsetLeft and offsetWidth instead of getLayoutBox() because
the correct measurements are not taken into account when using a CSS transform (such as translate),
which we are using in the i-amphtml-amp-story-shopping-tag-inner-flipped class.